packages/form-footer/src/FormFooter.spec.tsx

```diff
@@ -39,30 +46,119 @@ describe('packages/form-footer', () => {
+          primaryButton: { text: 'Test button' },
+        });
+        const Cancel = getByText('Cancel');
+        expect(Cancel).toBeInTheDocument();
+      });
+
+      test('Renders cancel button with custom text', () => {
+        const { getByText } = renderFooter({
+          primaryButton: { text: 'Test button' },
+          cancelButtonText: 'CancelText',
+        });
+        const Cancel = getByText('CancelText');
+        expect(Cancel).toBeInTheDocument();
+      });
+
+      test('Renders back button', () => {
+        const { getByText } = renderFooter({
+          primaryButton: { text: 'Test button' },
+          backButtonText: 'Back',
+        });
+        const Back = getByText('Back');
+        expect(Back).toBeInTheDocument();
       });
-      const Cancel = getByText('Cancel');
-      expect(Cancel).toBeInTheDocument();
     });
 
-    test('Renders cancel button with custom text', () => {
-      const { getByText } = renderFooter({
-        primaryButton: { text: 'Test button' },
-        cancelButtonText: 'CancelText',
+    describe('cancel button', () => {
+      test('renders with custom text if cancelButtonProps is defined', () => {
+        const cancelButtonText = 'Cancel';
+        const { queryByText } = renderFooter({
+          primaryButton: { text: 'Test button' },
+          cancelButtonProps: {
+            text: cancelButtonText,
+          },
+        });
+        const Cancel = queryByText(cancelButtonText);
+        expect(Cancel).toBeInTheDocument();
+      });
+
+      test('does not render if cancelButtonProps is not defined', () => {
+        const { queryByTestId } = renderFooter({
+          primaryButton: { text: 'Test button' },
+          cancelButtonText: '', // remove once deprecated props are removed
+        });
+        const Cancel = queryByTestId(buttonTestId.cancel);
+        expect(Cancel).not.toBeInTheDocument();
       });
-      const Cancel = getByText('CancelText');
-      expect(Cancel).toBeInTheDocument();
     });
 
-    test('Renders back button', () => {
-      const { getByText } = renderFooter({
-        primaryButton: { text: 'Test button' },
-        backButtonText: 'Back',
+    describe('back button', () => {
+      test('renders with custom text if backButtonProps is defined', () => {
+        const backButtonText = 'Back';
+        const { queryByText } = renderFooter({
+          primaryButton: { text: 'Test button' },
+          backButtonProps: {
+            text: backButtonText,
+          },
+        });
+        const Back = queryByText(backButtonText);
+        expect(Back).toBeInTheDocument();
+      });
+
+      test('does not render', () => {
+        const { queryByTestId } = renderFooter({
+          primaryButton: { text: 'Test button' },
+        });
+        const Back = queryByTestId(buttonTestId.back);
+        expect(Back).not.toBeInTheDocument();
+      });
+
+      describe('Back button left glyph', () => {
```

Comment:
nit: can omit `"Back button"` here since it's nested in another `describe` block